### ISSUE TITLE
conntrack shows wrong IP after changing the backend Pod of K8s ClusterIP service

### ISSUE THREAD

**Issue by subrnath** [kubernetes/kubernetes#82727]
**What happened**:

There are two Pods behind a K8s clusterIP service. It's expected only one Pod to carry the traffic. Hence label corresponding to the "selector" of the service is attached to first Pod. There are two client Pods scheduled on different worker nodes of the K8s cluster. TCP connection from those two client Pods established properly towards the ClusterIP of the K8s service. "conntrack -L" on both the worker nodes of the client Pods shows that NAT is properly changing the clusterIP of the k8s service towards the first Pod's calico IP.
 
Now i changed the label from backend first Pod to second Pod. Verified in all the worker nodes. Kubeproxy is updating the iptables rules properly from "iptables-save" command. But conntrack of one worker nodes shows wrong output. This issue is random. Sometimes seen in one worker nodes and some occurrences, it appears in both the worker nodes.

This issue is similar to https://github.com/kubernetes/kubernetes/issues/66651, https://github.com/kubernetes/kubernetes/pull/59286

Details

Client Pod-1 calico IP 10.244.236.74, worker1,
Client Pod-2 calico IP 10.244.182.178, worker2
K8s svc IP 10.111.190.183
back end Pods IP for the SVC is 10.244.169.72 ( first Pod ) ,10.244.200.50 ( second Pod)

Initially first Pod 10.244.169.72 is attached to the label.

$Kubectl get ep
SVC1 10.244.169.72:14101,10.244.169.72:14031,10.244.169.72:500

Worker-1
--------------
$sudo conntrack -L | grep -i 10.111.190.183 | grep -i 14101
tcp      6 299 ESTABLISHED src=10.244.236.74 dst=10.111.190.183 sport=14101 dport=14101 src=10.244.169.72 dst=10.244.236.74 sport=14101 dport=14101 [ASSURED] mark=0 use=1

Worker-2
$sudo conntrack -L | grep -i 10.111.190.183 | grep -i 14101
tcp      6 86399 ESTABLISHED src=10.244.182.178 dst=10.111.190.183 sport=14101 dport=14101 src=10.244.169.72 dst=10.244.182.178 sport=14101 dport=14101 [ASSURED] mark=0 use=1

All OK till this time.

Now change the label from first pod to second pod (10.244.200.50).
$Kubectl get ep
SVC1 10.244.200.50:14101,10.244.200.50:14031,10.244.200.50:500

Also verified iptables rules in all the worker nodes. Kubeproxy has updated properly.

Worker-1
$ sudo conntrack -L | grep -i 10.111.190.183 | grep -i 14101
tcp      6 86399 ESTABLISHED src=10.244.236.74 dst=10.111.190.183 sport=14101 dport=14101 src=10.244.200.50 dst=10.244.236.74 sport=14101 dport=14101 [ASSURED] mark=0 use=2

Worker-2
$ sudo conntrack -L | grep -i 10.111.190.183 | grep -i 14101
tcp      6 119 SYN_SENT src=10.244.182.178 dst=10.111.190.183 sport=14101 dport=14101 [UNREPLIED] src=10.111.190.183 dst=10.40.64.132 sport=14101 dport=15301 mark=0 use=1

Here worker-1 conntrack ok which is pointing to the backend Pod 10.244.200.50 but conntrack in woker-2 is wrong. "10.40.64.132" is the Node's IP.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

Two Pods behind a service. From application point of view, they are in active-standby protection mode. Label is attached to the first Pod which is the active Pod from app point of view. Do a switchover of the label from Pod-1 to Pod-2. Kubeproxy updates the Pod details in iptables properly but "conntrack -L" shows wrong entry in the worker nodes.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.15
- Cloud provider or hardware configuration: 
- OS (e.g: `cat /etc/os-release`): ubuntu 16.04.6 LTS (Xenial Xerus)
- Kernel (e.g. `uname -a`): ubuntu 4.10.0-27-generic
- Install tools:
- Network plugin and version (if this is a network-related bug): Calico v3.3.1
- Others:

/sig network 
/sig service-catalog
@kubernetes/sig-network-bugs
@kubernetes/sig-service-catalog-bugs

**Comment by athenabot** [comment]
/triage unresolved

Comment `/remove-triage unresolved` when the issue is assessed and confirmed.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by subrnath** [comment]
We identified the issue as when backend Pods are being switched, there is small window of  time where the iptables rules doesn’t have any entry. In that time, if the client retriggers the TCP SYN packets to the Cluster IP, it will not find any entry in iptables rule. So it will lookup routing table rule and take the default route. Due to that reason, conntrack shows the wrong entry which is not cleared even after kube-proxy updates the new Pod details in the iptables rule. If we manually flush the conntrack, it then works fine. We will provide a fix in app such that, before client triggers the request, check if the endpoint is updated.

Closing the issue.
